scripts/ci/sync_extensions.py

```diff
@@ -0,0 +1,141 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+# pylint: disable=broad-except
+
+import os
+import re
+import json
+from subprocess import check_output
+
+DEFAULT_TARGET_INDEX_URL = os.getenv('AZURE_EXTENSION_TARGET_INDEX_URL')
+STORAGE_ACCOUNT_KEY = os.getenv('AZURE_EXTENSION_TARGET_STORAGE_ACCOUNT_KEY')
+STORAGE_ACCOUNT = os.getenv('AZURE_EXTENSION_TARGET_STORAGE_ACCOUNT')
+STORAGE_CONTAINER = os.getenv('AZURE_EXTENSION_TARGET_STORAGE_CONTAINER')
+
+
+def _get_updated_extension_names():
+    cmd = 'git --no-pager diff --diff-filter=ACMRT HEAD~1 -- src/index.json'
+    updated_content = check_output(cmd.split()).decode('utf-8')
+    updated_urls = [line.replace('+', '') for line in updated_content.splitlines() if line.startswith('+') and not line.startswith('+++') and 'downloadUrl' in line]
```

Comment:
why not load the json content and get from key directly?

Reply:
Here we only extract updated downloadUrl from git diff for incremental sync.